engine/swarm/secrets.md

```diff
@@ -0,0 +1,812 @@
+    Remember that you cannot update or rename a secret, but you can revoke a
+    secret and grant access to it using a new target filename.
+
+    ```bash
+    $ docker service update --secret-rm mysql_password mysql
+
+    $ docker service update --secret-add source=mysql_password,target=old_mysql_password \
+                            --secret-add source=mysql_password_v2,target=mysql_password \
+                            mysql
+    ```
+
+    Updating a service causes it to restart, and when the MySQL service restarts
+    the second time, it has access to the old secret under
+    `/run/secrets/old_mysql_password` and the new secret under
+    `/run/secrets/mysql_password`.
+
+    Even though the MySQL service has access to both the old and new secrets
+    now, the MySQL root password has not yet been changed.
+
+3.  Now, change the MySQL root password using the `mysql` CLI. This command
+    reads the old and new password from the files in `/run/secrets` but does not
+    expose them on the command line or save them in the shell history.
+
+    Do this quickly and move on to the next step, because WordPress will lose
+    the ability to connect to MySQL.
+
+    First, find the ID of the `mysql` container task.
+
+    ```bash
+    $ docker ps |grep mysql |awk {'print $1'}
+
+    c7705cf6176f
+    ```
+
+    Substitute the container ID in the step below. Your container ID will be
+    different from `c7705cf6176f`.
+
+    ```bash
+    $ docker exec c7705cf6176f \
+             bash -c 'mysqladmin --user=root --password="$(< /run/secrets/old_mysql_password)" password "$(< /run/secrets/mysql_password)"'
+    ```
+
+4.  Update the `wordpress` service to use the new password, keeping the target
+    path at `/run/secrets/wp_db_secret` and keeping the file permissions at
+    `0400`.  This will trigger a rolling restart of the WordPress service and
+    the new secret will be used.
+
+    ```bash
+    $ docker service update wordpress \
+                            --secret-rm mysql_password \
+                            --secret-add source=mysql_password_v2,target=wp_db_password,mode=0400
+
+    wordpress
+
+    ```
+
+5.  Verify that WordPress works by browsing to http://localhost:30000/ on any
+    swarm node again. You'll need to use the WordPress username and password
+    from when you ran through the WordPress wizard in the previous task.
+
+6.  Revoke accsss to the old secret from the MySQL service and
+    remove the old secret from Docker.
+
+    ```bash
+
+    $ docker service update --secret-rm mysql_password mysql
+
+    $ docker secret remove mysql_password
+    ```
+
+
+7.  If you want to try the running all of these examples again or just want to
+    clean up after running through them, use these commands to remove the
+    WordPress service, the MySQL container, the Docker secrets, and the
+    `tempdir/` directory.
+
+    ```bash
+    $ docker service rm wordpress
+
+    $ docker rm -f mysql
```

Comment:
this will be `docker service rm mysql`, or combine with the previous one (`docker service rm wordpress mysql`)

also, the named volumes can be cleaned up afterwards